File: src/fiefapp.svelte/src/models/merchant.ts
import type { IPerson } from './person';

interface IMerchant {
  cargoId?: string;
  portId?: string;
  soldiers: string[];
  status: string;
}

export class Merchant implements IMerchant, IPerson {
  id: string;
  cargoId?: string;
  portId?: string;
  soldiers: string[];
  status: string;
  name: string;
  age: number;
  resources: number;
  loyalty: number;
  skill: number;

  constructor() {
    this.id = '';
    this.cargoId = undefined;
    this.portId = undefined;
    this.soldiers = [];
    this.status = '';
    this.name = '';
    this.age = 0;
    this.resources = 0;
    this.loyalty = 0;
    this.skill = 0;
  }
}

File: src/fiefapp.entities/Merchant.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Merchant : Person, IBaseEntity, IAssignable
    {
        public override Guid Id { get; set; }
        public override string Name { get; set; }
        public override int Age { get; set; }
        public override int Resources { get; set; }
        public override int Loyalty { get; set; }
        public override int Skill { get; set; }
        public override ICollection<string> Notes { get; set; }
        public Guid? AssignedTo { get; set; }
    }
}
